hs_tools_resource/forms.py

```diff
@@ -130,17 +130,20 @@ def __init__(self, allow_edit=True, res_short_id=None, element_id=None, *args, *
         super(SupportedResTypesForm, self).__init__(*args, **kwargs)
         self.fields['supported_res_types'].label = "Choose Resource Types:"
         self.helper = SupportedResTypeFormHelper(allow_edit, res_short_id, element_id, element_name='SupportedResTypes')
-        if self.instance:
-            supported_res_types = self.instance.supported_res_types.all()
-            if len(supported_res_types) > 0:
-                # NOTE: The following code works for SWAT res type but does not work here!!!
-                # self.fields['supported_res_types'].initial =
-                #   [parameter.description for parameter in supported_res_types]
-
-                self.initial['supported_res_types'] = \
-                    [parameter.description for parameter in supported_res_types]
-            else:
-                self.initial['supported_res_types'] = []
+        try:
```

Comment:
Instead of using ```try ... except```, do something like this.
Before the call to ```super()```,  store the value of instance from the kwargs
```model_instance = kwargs.get('instance')```

Then use  ``` if model_instance:```